scipy/optimize/nonlin.py

```diff
@@ -1350,7 +1350,7 @@ class KrylovJacobian(Jacobian):
         preconditioners. For example,
 
         >>> jac = BroydenFirst()
-        >>> kjac = KrylovJacobian(inner_M=jac.inverse).
+        >>> kjac = KrylovJacobian(inner_M=jac.inverse)
```

Comment:
No idea what was intended here, but the suggestion sounds reasonable. At least

 >>> kjac = optimize.nonlin.KrylovJacobian(inner_M=optimize.nonlin.InverseJacobian(jac))

doesn't give an error. This is not a public function, so maybe just make the change now and get this PR merged.